src/replays/_replayswidget.py

```diff
@@ -501,19 +501,24 @@ def onlineTreeClicked(self, item):
                     item.generateInfoPlayersHtml()
 
     def onlineTreeDoubleClicked(self, item):
-        if hasattr(item, "duration"):
+        if hasattr(item, "duration"):  # it's a game not a date separator
             if "playing" in item.duration:  # live game will not be in vault
-                if not item.live_delay:  # live game under 5min
-                    name = None
-                    if item.mod == "ladder1v1":
-                        name = item.name[:item.name.find(" ")]  # "name vs name"
-                    else:
-                        players = (p for t in item.teams for p in t)
-                        for player in players:
-                            name = player["name"]
-                            if name != "" and name in self._playerset:
+                # search result isn't updated automatically - so game status might have changed
+                if item.uid in client.instance.gameset.games:  # game still running
```

Comment:
I'd rather we passed gameset through the constructor the same way we do with playerset.

Reply:
you mean add gameset to ReplayVaultWidgetHandler ?

Reply:
Yeah, the same way playerset is added. We don't even have to look far for it, since ReplaysWidget has it.

Reply:
ok, done.
(but you recently wrote - when I asked about 'self.....X' visa 'client.instant...X' - that it doesn't matter)

Reply:
Client is a shoggoth class that we can't hope to make manageable, so it doesn't matter that much. These however are shiny new model classes, so proper dependency injection is very much in fashion :)